lib/src/google_sign_in_mocks_base.dart

```diff
@@ -4,13 +4,22 @@ import 'package:mockito/mockito.dart';
 class MockGoogleSignIn extends Mock implements GoogleSignIn {
   MockGoogleSignInAccount _currentUser;
 
+  /// To simulate google login cancellation behaviour.
+  bool _isCancelled = false;
+
+
+  void setIsCancelled(bool val) {
+    _isCancelled = val;
+  }
   @override
   GoogleSignInAccount get currentUser => _currentUser;
 
   @override
   Future<GoogleSignInAccount> signIn() {
     _currentUser = MockGoogleSignInAccount();
-    return Future.value(_currentUser);
+    final returnVal = _isCancelled ? Future.value(null) : Future.value(_currentUser);
+    _isCancelled = false;
```

Comment:
It'd be easier to document and for users to understand if signIn simply returned a value based on _isCancelled, but didn't change _isCancelled itself. Can you remove this line?

Reply:
I guess _isCancelled must be set back to false, because it may cause conflicts with other tests.

If the _isCancelled was not changed, the third test will fail.
`
import 'package:google_sign_in_mocks/google_sign_in_mocks.dart';
import 'package:test/test.dart';

void main() {
  MockGoogleSignIn googleSignIn;
  setUpAll((){
    googleSignIn = MockGoogleSignIn();
  });

  test('should return idToken and accessToken when authenticating', () async {
    final signInAccount = await googleSignIn.signIn();
    final signInAuthentication = await signInAccount.authentication;
    expect(signInAuthentication, isNotNull);
    expect(googleSignIn.currentUser, isNotNull);
    expect(signInAuthentication.accessToken, isNotNull);
    expect(signInAuthentication.idToken, isNotNull);
  });

  test('should return null when google login is cancelled by the user', () async {
    googleSignIn.setIsCancelled(true);
    final signInAccount = await googleSignIn.signIn();
    expect(signInAccount, isNull);
  });

  test('shouldn\'t return null when google login is cancelled by the user', () async {
    final signInAccount = await googleSignIn.signIn();
    expect(signInAccount, isNotNull);
  });
}
`

Reply:
If users want each test to be independent, they should use setUp, which would set up a new instance of `GoogleSignIn` before each test.

If they really want to test signing in twice (once canceled, once not canceled), users should be doing it like so:
```dart
    googleSignIn.setIsCancelled(true);
    final signInAccount = await googleSignIn.signIn();
    expect(signInAccount, isNull);
    googleSignIn.setIsCancelled(true);
    final signInAccountSecondAttempt = await googleSignIn.signIn();
    expect(signInAccountSecondAttempt, isNotNull);
```

Would you mind removing this line (`_isCancelled = false`)?

https://api.flutter.dev/flutter/flutter_test/setUp.html

Reply:
I forgot to use `setUp` for that, sorry. I've removed the line.